go/libraries/doltcore/sqle/dprocedures/dolt_branch.go

```diff
@@ -15,16 +15,314 @@
+	if err == nil {
+		if repoState, err := env.LoadRepoState(fs); err == nil {
+			rs = repoState
+			headOnCLI = repoState.Head.Ref.GetPath()
+		}
+	}
+
+	// Verify that we can delete all branches before continuing
+	for _, branchName := range apr.Args {
+		if err = branch_control.CanDeleteBranch(ctx, branchName); err != nil {
+			return err
+		}
+	}
+
+	var updateFS = false
+	for _, branchName := range apr.Args {
+		if len(branchName) == 0 {
+			return EmptyBranchNameErr
+		}
+		force := apr.Contains(cli.DeleteForceFlag) || apr.Contains(cli.ForceFlag)
+
+		if !force {
+			err = validateBranchNotActiveInAnySession(ctx, branchName)
+			if err != nil {
+				return err
+			}
+		}
+		err = actions.DeleteBranch(ctx, dbData, loadConfig(ctx), branchName, actions.DeleteOptions{
+			Force: force,
+		})
+		if err != nil {
+			return err
+		}
+		if headOnCLI == branchName {
+			updateFS = true
+		}
+	}
+
+	if fs != nil && updateFS {
+		rs.Head.Ref = ref.NewBranchRef("")
+		rs.Save(fs)
+	}
+
+	return nil
+}
+
+// validateBranchNotActiveInAnySessions returns an error if the specified branch is currently
+// selected as the active branch for any active server sessions.
+func validateBranchNotActiveInAnySession(ctx *sql.Context, branchName string) error {
+	currentDbName, _, err := getRevisionForRevisionDatabase(ctx, ctx.GetCurrentDatabase())
+	if err != nil {
+		return err
+	}
+
+	if currentDbName == "" {
+		return nil
+	}
+
+	if sqlserver.RunningInServerMode() == false {
+		return nil
+	}
+
+	runningServer := sqlserver.GetRunningServer()
+	if runningServer == nil {
+		return nil
+	}
+	sessionManager := runningServer.SessionManager()
+	branchRef := ref.NewBranchRef(branchName)
+
+	return sessionManager.Iter(func(session sql.Session) (bool, error) {
+		dsess, ok := session.(*dsess.DoltSession)
+		if !ok {
+			return false, fmt.Errorf("unexpected session type: %T", session)
+		}
+
+		sessionDatabase := dsess.Session.GetCurrentDatabase()
+		sessionDbName, _, err := getRevisionForRevisionDatabase(ctx, dsess.GetCurrentDatabase())
+		if err != nil {
+			return false, err
+		}
+
+		if len(sessionDatabase) == 0 || sessionDbName != currentDbName {
+			return false, nil
+		}
+
+		activeBranchRef, err := dsess.CWBHeadRef(ctx, sessionDatabase)
+		if err != nil {
+			return false, err
+		}
+
+		if ref.Equals(branchRef, activeBranchRef) {
+			return false, fmt.Errorf("unsafe to delete or rename branches in use in other sessions; " +
+				"use --force to force the change")
+		}
+
+		return false, nil
+	})
+}
+
+func loadConfig(ctx *sql.Context) *env.DoltCliConfig {
+	// When executing branch actions from SQL, we don't have access to a DoltEnv like we do from
+	// within the CLI. We can fake it here enough to get a DoltCliConfig, but we can't rely on the
+	// DoltEnv because tests and production will run with different settings (e.g. in-mem versus file).
+	dEnv := env.Load(ctx, env.GetCurrentUserHomeDir, filesys.LocalFS, doltdb.LocalDirDoltDB, "")
```

Comment:
So I traced this down and the only reason we need this is to do this:

```go
		ms, err := doltdb.NewCommitSpec(env.GetDefaultInitBranch(config))
```

Super lame use of resources. Can you add a TODO to fix this with the note that we super don't need an entire client config here? (It should be available in the server context in some other way)